internal/terraform/module/module_ops.go

```diff
@@ -166,14 +166,76 @@ func ParseConfiguration(fs filesystem.Filesystem, modStore *state.ModuleStore, m
 		if !strings.HasSuffix(name, ".tf") || IsIgnoredFile(name) {
 			continue
 		}
-
 		// TODO: overrides
 
 		fullPath := filepath.Join(modPath, name)
 
 		src, err := fs.ReadFile(fullPath)
 		if err != nil {
-			sErr := modStore.UpdateParsedFiles(modPath, files, err)
+			sErr := modStore.UpdateParsedModuleFiles(modPath, files, err)
+			if sErr != nil {
+				return sErr
+			}
+			return err
+		}
+
+		f, pDiags := hclsyntax.ParseConfig(src, name, hcl.InitialPos)
+		diags[name] = pDiags
+		if f != nil {
+			files[name] = f
+		}
+	}
+
+	sErr := modStore.UpdateParsedModuleFiles(modPath, files, err)
+	if sErr != nil {
+		return sErr
+	}
+
+	sErr = modStore.UpdateModuleDiagnostics(modPath, diags)
+	if sErr != nil {
+		return sErr
+	}
+
+	return err
+}
+
+func ParseVariables(fs filesystem.Filesystem, modStore *state.ModuleStore, modPath string) error {
+	err := modStore.SetVarsParsingState(modPath, op.OpStateLoading)
+	if err != nil {
+		return err
+	}
+
+	files := make(map[string]*hcl.File, 0)
+	diags := make(map[string]hcl.Diagnostics, 0)
+
+	infos, err := fs.ReadDir(modPath)
+	if err != nil {
+		sErr := modStore.UpdateParsedVarsFiles(modPath, files, err)
+		if sErr != nil {
+			return sErr
+		}
+		return err
+	}
+
+	for _, info := range infos {
+		if info.IsDir() {
+			// We only care about files
+			continue
+		}
+
+		name := info.Name()
+		if !(strings.HasSuffix(name, ".auto.tfvars") ||
+			strings.HasSuffix(name, ".auto.tfvars.json") ||
+			name == "terraform.tfvars" ||
+			name == "terraform.tfvars.json") || IsIgnoredFile(name) {
```

Comment:
```suggestion
		if !(strings.HasSuffix(name, ".auto.tfvars") ||
			name == "terraform.tfvars") || IsIgnoredFile(name) {
```

I think the JSON is better to be addressed in a separate PR - I'm not even sure what exact implications this should have from user perspective yet.

Also we'll need to double check that all upstream decoders can deal with JSON files - although the parser always returns `*hcl.File` for both "native" HCL and JSON, they have different structure and AFAICT we cannot use `hclsyntax` package at all for parsed JSON files.

I assume that eventually we'll want to show values + depending on Terraform's own behaviour we may want to report duplicate variable names across auto-loaded files. Either way I'd leave this as explicitly unsupported for now.